config/config.go

```diff
@@ -101,12 +118,30 @@ func (c *Config) Set(key, value string) error {
 		return fmt.Errorf("invalid key %s", key)
 	}
 
+	// If read configList after open with O_TRUNC, config file has been truncated to be empty.
+	// Therefore we have to open another file descriptor to read configList.
+	configFile, err := os.Open(c.filePath)
+	if err != nil {
+		return err
+	}
+	defer configFile.Close()
+
+	var configList = configList{}
+
+	err = yaml.NewDecoder(configFile).Decode(&configList)
+	if err != nil {
+		return fmt.Errorf("failed to parse config file: %v", err)
+	}
```

Comment:
How about to create a function which get config setting from file?

Reply:
I implement a such function.